src/PKSim.Core/Snapshots/Mappers/ObserverMapper.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System.Threading.Tasks;
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Core.Domain.Descriptors;
+using OSPSuite.Core.Services;
+using OSPSuite.Utility.Extensions;
+using PKSim.Assets;
+using PKSim.Core.Repositories;
+using SnapshotObserver = PKSim.Core.Snapshots.Observer;
+using ModelObserver = OSPSuite.Core.Domain.Builder.IObserverBuilder;
+
+namespace PKSim.Core.Snapshots.Mappers
+{
+   public class ObserverMapper : ObjectBaseSnapshotMapperBase<ModelObserver, SnapshotObserver>
+   {
+      private const string AMOUNT_OBSERVER = "Amount";
+      private const string CONTAINER_OBSERVER = "Container";
+
+      private readonly DescriptorConditionMapper _descriptorConditionMapper;
+
+      private readonly ExplicitFormulaMapper _explicitFormulaMapper;
+
+      private readonly MoleculeListMapper _moleculeListMapper;
+      private readonly IObjectBaseFactory _objectBaseFactory;
+      private readonly IDimensionRepository _dimensionRepository;
+      private readonly ILogger _logger;
+
+      public ObserverMapper(
+         DescriptorConditionMapper descriptorConditionMapper, 
+         ExplicitFormulaMapper explicitFormulaMapper, 
+         MoleculeListMapper moleculeListMapper,
+         IObjectBaseFactory objectBaseFactory,
+         IDimensionRepository dimensionRepository,
+         ILogger logger)
+      {
+         _descriptorConditionMapper = descriptorConditionMapper;
+         _explicitFormulaMapper = explicitFormulaMapper;
+         _moleculeListMapper = moleculeListMapper;
+         _objectBaseFactory = objectBaseFactory;
+         _dimensionRepository = dimensionRepository;
+         _logger = logger;
+      }
+
+      public override async Task<SnapshotObserver> MapToSnapshot(ModelObserver observer)
+      {
+         var snapshot = await SnapshotFrom(observer);
+         snapshot.Dimension = SnapshotValueFor(observer.Dimension?.Name);
+         snapshot.ContainerCriteria = await _descriptorConditionMapper.MapToSnapshots(observer.ContainerCriteria);
+
+         //We do not support any other type of formula at the moment
+         snapshot.Formula = await _explicitFormulaMapper.MapToSnapshot(observer.Formula as OSPSuite.Core.Domain.Formulas.ExplicitFormula);
```

Comment:
what will happen then in case of other formula types (table etc.)? MapToSnapshot will crash then?

Reply:
Yes and I hope it will crash since we only support ExplicitFormula. We need to catch it as a BIG fat exception